src/sc/proofverifier.cpp

```diff
@@ -58,241 +195,349 @@ void CScProofVerifier::LoadDataForCswVerification(const CCoinsViewCache& view, c
-            custom_fields[i] = sptrFe.get();
-            vSptr.push_back(sptrFe);
-            i++;
-        }
-
-        const backward_transfer_t* bt_list_ptr = input.bt_list.data();
-        int bt_list_len = input.bt_list.size();
-
-        // mc crypto lib wants a null ptr if we have no fields
-        if (custom_fields_len == 0)
-            custom_fields.reset();
-        if (bt_list_len == 0)
-            bt_list_ptr = nullptr;
-
-        wrappedFieldPtr   sptrConst  = input.constant.GetFieldElement();
-        wrappedFieldPtr   sptrCum    = input.endEpochCumScTxCommTreeRoot.GetFieldElement();
-        wrappedScProofPtr sptrProof  = input.certProof.GetProofPtr();
-        wrappedScVkeyPtr  sptrCertVk = input.CertVk.GetVKeyPtr();
-
-        bool ret = batchVerifier.add_certificate_proof(
-            idx,
-            sptrConst.get(),
-            input.epochNumber,
-            input.quality,
-            bt_list_ptr,
-            bt_list_len,
-            custom_fields.get(),
-            custom_fields_len,
-            sptrCum.get(),
-            input.mainchainBackwardTransferRequestScFee,
-            input.forwardTransferScFee,
-            sptrProof.get(),
-            sptrCertVk.get(),
-            &code
-        );
-        idx++;
-
-        //dumpFeArr((field_t**)custom_fields.get(), custom_fields_len, "custom fields");
-        //dumpFe(sptrCum.get(), "cumTree");
+            CProofVerifierItem& item = proof.second;
 
-        if (!ret || code != CctpErrorCode::OK)
-        {
-            LogPrintf("ERROR: %s():%d - cert [%s] has proof which does not verify: ret[%d], code [0x%x]\n",
-                __func__, __LINE__, input.certHash.ToString(), (int)ret, code);
+            // Set as "passed" only proofs that didn't fail during the add process.
+            if (item.result == ProofVerificationResult::Unknown)
+            {
+                item.result = ProofVerificationResult::Passed;
+            }
         }
     }
-
-    CZendooBatchProofVerifierResult verRes(batchVerifier.batch_verify_all(&code));
-
-    if (!verRes.Result())
+    else
     { 
-        LogPrintf("ERROR: %s():%d - verify failed for %d proof(s), code [0x%x]\n",
+
+        if (verRes.FailedProofs().size() > 0)
+        {
+            // We know for sure that some proofs failed, the other ones may be valid or not (unknown).
+            LogPrintf("ERROR: %s():%d - verify failed for %d proof(s), code [0x%x]\n",
             __func__, __LINE__, verRes.FailedProofs().size(), code);
-        return false; 
+
+            for (uint32_t proofId : verRes.FailedProofs())
+            {
+                uint256 txHash = proofIdMap.at(proofId);
+                proofs.at(txHash).result = ProofVerificationResult::Failed;
+            }
+        }
+        else
+        {
+            // We don't know which proofs made the batch process fail.
+            LogPrintf("ERROR: %s():%d - verify failed without detailed information\n", __func__, __LINE__);
```

Comment:
Is it meaningful printing also the error code here?

Reply:
Fixed.